pkg/shim/scheduler_mock_test.go

```diff
@@ -128,6 +128,23 @@ func (fc *MockScheduler) addNode(nodeName string, nodeLabels map[string]string,
 	return fc.apiProvider.GetAPIs().SchedulerAPI.UpdateNode(request)
 }
 
+func (fc *MockScheduler) waitApplicationDeleted(t *testing.T, appID string) {
+	deadline := time.Now().Add(10 * time.Second)
+	for {
+		app := fc.context.GetApplication(appID)
+		if app == nil {
+			break
+		}
+		log.Log(log.Test).Info("waiting for app deleted",
+			zap.String("appID", appID))
+		time.Sleep(time.Second)
+		if time.Now().After(deadline) {
+			t.Errorf("application %s is not deleted in given time", appID)
+			return
+		}
+	}
```

Comment:
I know it's probably based on `waitAndAssertApplicationState()`, but there's a better way. Simply use `utils.WaitForCondition()`.

```
err := utils.WaitForCondition(func() bool {
		return fc.context.GetApplication(appID) != nil
	}, time.Second, 5*time.Second)
assert.NilError(t, err, "application has not been deleted")
```

Reply:
Addressed in latest PR, thanks @pbacsko for review.